### PR TITLE
Port get-process

### PR DESCRIPTION
**Pull request by sieut**
I'm trying to port get-process and I've looked at other porting PRs/issues to learn from examples how to do it. With what I have so far, I got an error while compiling

```
mkdir -p ../rust_src/target/
echo " " > ../rust_src/target/.rustflags.tmp
diff -q ../rust_src/target/.rustflags ../rust_src/target/.rustflags.tmp || cp ../rust_src/target/.rustflags.tmp ../rust_src/target/.rustflags
rm -f ../rust_src/target/.rustflags.tmp
make -C ../admin/charsets all
make[2]: Entering directory '/home/sieut/work/remacs/admin/charsets'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/sieut/work/remacs/admin/charsets'
make -C ../admin/unidata charscript.el
make[2]: Entering directory '/home/sieut/work/remacs/admin/unidata'
make[2]: Nothing to be done for 'charscript.el'.
make[2]: Leaving directory '/home/sieut/work/remacs/admin/unidata'
  CCLD     temacs
.././rust_src/target/debug/libremacs.a(remacs-d34508946e55adfd.0.o): In function `remacs::process::get_process':
/home/sieut/work/remacs/rust_src/src/process.rs:(.text._ZN6remacs7process11get_process17hb01f2e359f17c0afE+0x7d): undefined reference to `Vprocess_alist'
collect2: error: ld returned 1 exit status
Makefile:624: recipe for target 'temacs' failed
make[1]: *** [temacs] Error 1
```

I'm not sure what causes this error and how to fix it. I'd love to have some hints or explanation so that I can continue on the issue.

### CODE REVIEW

**Comment by DavidDeSimone** [comment]
So what you are seeing is a linker error. Emac's build process is kind of convoluted, but creating temacs is where we link together the C and Rust. If there are any extern variables that the linker fails to find, this is the part that will fail.

So in rust, you are declaring an extern variable called "Vprocess_alist". The problem here is that "Vprocess_alist" is declared "static" in process.c. This means that Vprocess-alist will be limited to the compilation unit of process.c. Try removing the "static" in front of the declaration of Vprocess-alist in process.c

**Comment by sieut** [comment]
Thank you @DavidDeSimone, removing static from Vprocess_alist did get rid of the error. I used the function in ielm and it seems to be working, I'm gonna take the WIP tag off of the title

File: rust_src/src/process.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//! Functions operating on process
+
+use remacs_macros::lisp_fn;
+use remacs_sys::Vprocess_alist;
+use lisp::LispObject;
+use lists::{assoc, cdr};
+
+///Return the process named NAME, or nil if there is none
```
Comment (birkenfeld) [inline rust_src/src/process.rs]: Missing a space before the docstring, and the period at the end.

**Comment by sieut** [comment]
Thank you, I accidentally checked in my tags file. I have removed it and fixed the docstrings as well.

**Comment by Wilfred** [comment]
Awesome! I've resolve the conflicts, so we can merge this once Travis is happy.

(If you fancy rebasing on master, that's fine too. I didn't realise that fixing the conflicts from the web UI would create a merge commit.)

**Pull request merged**